win32ss/user/user32/windows/dialog.c

```diff
@@ -2624,7 +2624,17 @@ IsDialogMessageW(
              if (!(dlgCode & DLGC_WANTARROWS))
              {
                  BOOL fPrevious = (lpMsg->wParam == VK_LEFT || lpMsg->wParam == VK_UP);
-                 HWND hwndNext = GetNextDlgGroupItem( hDlg, lpMsg->hwnd, fPrevious );
+
+                 /* Skip STATIC elements when arrow-moving through a list of controls */
+                 HWND hwndNext, hwndFirst = lpMsg->hwnd;
+                 for (hwndNext = GetNextDlgGroupItem(hDlg, hwndFirst, fPrevious);
+                      hwndNext && hwndFirst != hwndNext;
+                      hwndNext = GetNextDlgGroupItem(hDlg, hwndNext, fPrevious))
+                  {
+                      if (!(SendMessageW(hwndNext, WM_GETDLGCODE, 0, 0) & DLGC_STATIC))
+                          break;
+                  }
```

Comment:
Simpler/clearer:
```suggestion
                 HWND hwndFirst = lpMsg->hwnd, hwndNext = hwndFirst;
                 do
                 {
                     hwndNext = GetNextDlgGroupItem(hDlg, hwndNext, fPrevious);
                 }
                 while (hwndNext && hwndNext != hwndFirst &&
                        (SendMessageW(hwndNext, WM_GETDLGCODE, 0, 0) & DLGC_STATIC));
```

Reply:
Not sure that having a huge condition that contains the action to do (SendMessage) for a given item (hwndNext) makes anything clearer...

Reply:
Well, different points of view...
I would say `SendMessageW()` is a condition while `GetNextDlgGroupItem()` is the intended action.
Even your way,
```c
                 while ((hwndNext = GetNextDlgGroupItem(hDlg, hwndNext, fPrevious)) != NUL &&
                        hwndNext != hwndFirst)
```
would be simpler.